guides/keras_cv/object_detection_keras_cv.py

```diff
@@ -0,0 +1,848 @@
+parameter to highlight what class each box was assigned to.  Let's assemble a
+class mapping now.
+"""
+
+class_ids = [
+    "Aeroplane",
+    "Bicycle",
+    "Bird",
+    "Boat",
+    "Bottle",
+    "Bus",
+    "Car",
+    "Cat",
+    "Chair",
+    "Cow",
+    "Dining Table",
+    "Dog",
+    "Horse",
+    "Motorbike",
+    "Person",
+    "Potted Plant",
+    "Sheep",
+    "Sofa",
+    "Train",
+    "Tvmonitor",
+    "Total",
+]
+class_mapping = dict(zip(range(len(class_ids)), class_ids))
+
+"""
+Just like any other `keras.Model` you can predict bounding boxes using the
+`model.predict()` API.
+"""
+
+y_pred = pretrained_model.predict(image_batch)
+# y_pred is a bounding box Tensor:
+# {"classes": ..., boxes": ...}
+visualization.plot_bounding_box_gallery(
+    image_batch,
+    value_range=(0, 255),
+    rows=1,
+    cols=1,
+    y_pred=y_pred,
+    scale=5,
+    font_scale=0.7,
+    bounding_box_format="xywh",
+    class_mapping=class_mapping,
+)
+
+"""
+In order to support easy this easy and intuitive inference workflow, KerasCV
+performs non-max suppression inside of the `RetinaNet` class.
+Non-max suppression is a traditional computing algorithm that solves the problem
+of a model detecting multiple boxes for the same object.
+
+**Example of pre-NMS detections (left) and post-NMS detections (right):**
+
+<img width="400" src="https://i.imgur.com/L2KPpSF.jpg">
+
+Non-max suppression is a highly configurable algorithm, and in most cases you
+will want to customize the settings of your model's non-max
+suppression operation.
+This can be done by overriding to the `model.prediction_decoder` attribute.
+
+Let's use a custom `keras_cv.layers.MultiClassNonMaxSuppression` instance
+to perform prediction decoding in our pretrained model.
+In this case, we will tune the `iou_threshold` to `0.2`, and the
+`confidence_threshold` to `0.97`.
+
+Raising the `confidence_threshold` will cause the model to only output boxes
+that have a higher confidence score.  `iou_threshold` controls the threshold of
+IoU two boxes must have in order for one to be pruned out.
+[More information on these parameters may be found in the TensorFlow API docs](https://www.tensorflow.org/api_docs/python/tf/image/combined_non_max_suppression)
+"""
+
+prediction_decoder = keras_cv.layers.MultiClassNonMaxSuppression(
+    bounding_box_format="xywh",
+    from_logits=True,
+    # Decrease the required threshold to make predictions get pruned out
+    iou_threshold=0.2,
+    # Tune confidence threshold for predictions to pass NMS
+    confidence_threshold=0.7,
+)
+pretrained_model.prediction_decoder = prediction_decoder
+
+y_pred = pretrained_model.predict(image_batch)
+visualization.plot_bounding_box_gallery(
+    image_batch,
+    value_range=(0, 255),
+    rows=1,
+    cols=1,
+    y_pred=y_pred,
+    scale=5,
+    font_scale=0.7,
+    bounding_box_format="xywh",
+    class_mapping=class_mapping,
+)
+
+"""
+That looks a lot better!
```

Comment:
This is quite confusing because NMS is set up as *suppressing* noisy detections (false positives) and instead it is recovering true positives. Is this a good example?

Reply:
Updated the picture to reference the doc picture instead!  Thanks!